cli/azd/resources/apphost/templates/resources.bicept

```diff
@@ -345,6 +345,16 @@ resource {{bicepName $name}}OperatorRoleAssignment 'Microsoft.Authorization/role
     roleDefinitionId:  subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '230815da-be43-4aae-9cb4-875f7bd000aa')
   }
 }
+// Cosmos DB Contributor
+resource {{bicepName $name}}ContributorRoleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid({{bicepName $name}}.id, managedIdentity.id, subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '00000000-0000-0000-0000-000000000002'))
```

Comment:
I'm not certain that this is correct.  Per: https://learn.microsoft.com/en-us/azure/cosmos-db/how-to-setup-rbac it seems like these IDs are not to be used with traditional role assignments but instead are special cosmos specific things.

See https://learn.microsoft.com/en-us/azure/cosmos-db/how-to-setup-rbac#using-bicepazure-resource-manager-templates for some examples and you can see how the IaC is subtly different...

Reply:
yeah, I just tested it and it did not work :)  I was just experimenting here. Lol.

It helped me to build this other approach:

```bicep
var roleDefinitionId = guid('sql-role-definition-', managedIdentity.id, mycosmos.id)
var roleAssignmentId = guid(roleDefinitionId, managedIdentity.id, mycosmos.id)
resource mycosmos 'Microsoft.DocumentDB/databaseAccounts@2023-11-15' = {
  name: replace('mycosmos-${resourceToken}', '-', '')
  location: location
  tags: union(tags, {'aspire-resource-name': 'mycosmos'})
  properties: {
    consistencyPolicy: { defaultConsistencyLevel: 'Session' }
    locations: [
      {
        locationName: location
        failoverPriority: 0
      }
    ]
    databaseAccountOfferType: 'Standard'
  }

  resource roleDef 'sqlRoleDefinitions@2023-11-15' = {
    name: roleDefinitionId
    properties: {
      roleName: 'mycosmosRoleDefinition'
      type: 'CustomRole'
      assignableScopes: [
        mycosmos.id
      ]
      permissions: [
        {
          dataActions: [
            'Microsoft.DocumentDB/databaseAccounts/readMetadata'
            'Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers/*'
          ]
        }
      ]
    }
  }
  resource mycosmosCustomRoleAssignment 'sqlRoleAssignments@2023-11-15' = {
  name: roleAssignmentId
  properties: {
    principalId: managedIdentity.properties.principalId
    roleDefinitionId: roleDef.id
    scope: mycosmos.id
  }
}
}
```

Which works and creates a custom role and assigns it to the managed identity...  

However, on the Aspire side, this is not working because their `CosmosClient` is using data plane SDK and fails with:

```
Reason: (Request blocked by Auth mycosmosmyowqjn7m5qiu : The given request [POST /dbs] cannot be authorized by AAD token in data plane. Learn more: https://aka.ms/cosmos-native-rbac.
```

So, we might need to switch to use a connection string with a key, instead of rbac. Or wait for https://github.com/dotnet/aspire/issues/1546

Reply:
I don't think we should switch to using connection string. I think we should do this instead:

https://github.com/Azure/azure-dev/issues/3181

PR for the Aspire bits is already done:

https://github.com/dotnet/aspire/pull/1583

Reply:
It's worth noting that a custom role for general contributor access (what you've essentially created manually) is not required, instead you can use the `roleDefinitionId` of `00000000-0000-0000-0000-000000000002`.

Reply:
What you outlined here makes sense, @mitchdenny.  We'll do this.